core/trie2/triedb/pathdb/layertree.go

```diff
@@ -0,0 +1,196 @@
+package pathdb
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2/trienode"
+	"github.com/NethermindEth/juno/core/trie2/trieutils"
+)
+
+// Represents a layer of the layer tree
+type layer interface {
+	// Returns the encoded node bytes for a given trie id, owner, path and isLeaf flag
+	node(id trieutils.TrieID, owner *felt.Felt, path trieutils.Path, isLeaf bool) ([]byte, error)
+	// Updates the layer with a new root hash, state id and block number
+	update(root felt.Felt, id, block uint64, nodes *nodeSet) *diffLayer
+	// Writes the journal to the given writer
+	journal(w io.Writer) error
+	// Returns the root hash of the layer
+	rootHash() felt.Felt
+	// Returns the state id of the layer
+	stateID() uint64
+	// Returns the parent layer of the current layer
+	parentLayer() layer
+}
+
+// Represents a layer tree which contains multiple in-memory diff layers and a single disk layer
```

Comment:
I don't see any code in this structure that checks that there is only one disk layer. We should either give guarantee that it's only going to have a disk layer or update the description, as the expected usage, expects a single disk layer

Reply:
comment updated